cumulus_library/template_sql/sql_utils.py

```diff
@@ -209,7 +209,12 @@ def validate_schema(
     validated_schema = {}
     for table, cols in expected_table_cols.items():
         query = base_templates.get_column_datatype_query(schema, table, cols.keys())
-        table_schema = cursor.execute(query).fetchall()
+        try:
+            table_schema = cursor.execute(query).fetchall()
+        except Exception:
```

Comment:
i know this is a pain, but: can we catch a more specific error here? Two cases that should error out here are 'something happened to your network connection' and 'the query you sent was malformed in some way'.

Reply:
OK I can look into which specific error it is - probably a generic Athena error that I need to inspect the code of.

However - that's injecting some database-specific knowledge into this generic code. I could maybe add a method like "tell me if this exception is SomeKindOfError" to DatabaseBackend? Or just live with the boundary bleeding.

Reply:
i the the 'put it in the databasebackend' approach is probably the right one, but if it turns into A Whole Thing we could live with it for now and revisit.

Reply:
I'll try that approach and see

Reply:
OK added a new commit for easier reading.

I did three things:
- Unrelated syntax fix for an excess comma that I noticed
- Changed all the `core` builders to just grab `config` in `prepare_queries` so they could get direct access to the database backend.
- Added a DatabaseBackend.operational_errors() method to surface which exception classes are "operational errors" -- this is the abstraction layer I happened to choose, I'm not rooting for it hard and am open to push back (i.e. I could imagine a very specific method like "is_table_not_found_error()" or a generic method like "exception_to_error_enum()" - but this just an easy & quick path for now). In Athena, network errors and syntax errors would both be `DatabaseError`. There's also stuff like `ProgrammingError` or `NotSupportedError`. `OperationalError` seems to be used in both pyAthena and DuckDB for stuff like "out of memory" or "table already exists" or "table doesn't exist" kind of stuff.